### ISSUE TITLE
TypeError since last update (The "fd" argument must be of type number.)

### ISSUE THREAD

**Issue by Shirobaka** [Natizyskunk/vscode-sftp#34]
**Do you read the FAQ?**
- [X] Yes.
- [x] [I am going to read now.](https://github.com/Natizyskunk/vscode-sftp/blob/master/FAQ.md)

**Describe the bug**
Since the last update downloading won't work anymore

**To Reproduce**
With downloadOnOpen on true:
1. Open the File
This happens too when you try to download a file manually

**Expected behavior**
File should be downloadable

**Screenshots**
https://nextcloud.kanjiku.net/index.php/s/mKXMMqY4HRp9QSH/preview

**Desktop (please complete the following information):**
 - OS: Microsoft Windows 10 (10.0.19042 Build 19042)
 - VSCode Version 1.59.1
 - Extension Version 1.15.0

**Extension Logs from Startup** - *required*
https://pastebin.com/3CqA1aGa

**Comment by mrjcgoodwin** [comment]
Yep getting this too and extension crashing on loading VS code.

**Comment by codivist** [comment]
Also getting this when I right click on a folder and click on `Sync Remote -> local`
as well as right clicking on a file and clicking `Download`

I get the following error in the output tab:
```
[08-23 09:56:39] [warn] Can't set modified time to the file because The "fd" argument must be of type number. Received undefined
[08-23 09:56:39] [error] TypeError [ERR_INVALID_ARG_TYPE]: The "fd" argument must be of type number. Received undefined
	at Object.close (fs.js:417:3)
	at Object.close (/Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/graceful-fs/graceful-fs.js:50:23)
	at Object.t (/Users/user/.vscode/extensions/bmewburn.vscode-intelephense-client-1.7.1/lib/extension.js:1:24532)
	at Object.e.exports.close (/Users/user/.vscode/extensions/shan.code-settings-sync-3.4.3/out/extension.js:1:4112)
	at /Users/user/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:1:24769
	at new Promise (<anonymous>)
	at Object.close (/Users/user/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:1:24669)
	at s.close (/Users/user/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:61:67116)
	at t.default.<anonymous> (/Users/user/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:61:2880)
	at Generator.throw (<anonymous>)
	at a (/Users/user/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:61:583)
	at processTicksAndRejections (internal/process/task_queues.js:97:5) when remote ➞ local /Users/user/code/dev6/application/models/Cart/Paypal.php
```

**Comment by Mr-KayJayDee** [comment]
Also getting this error with the same error message as everyone here

**Comment by JoelLisenby** [comment]
Same error is occuring for me when downloading a *.log file.

```
[08-23 12:01:28] [error] TypeError [ERR_INVALID_ARG_TYPE]: The "fd" argument must be of type number. Received undefined
	at Object.close (fs.js:426:3)
	at Object.close (/Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/fs-extra/node_modules/graceful-fs/graceful-fs.js:54:23)
	at /Users/joel/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:1:24769
	at new Promise (<anonymous>)
	at Object.close (/Users/joel/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:1:24669)
	at s.close (/Users/joel/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:61:67116)
	at t.default.<anonymous> (/Users/joel/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:61:2880)
	at Generator.throw (<anonymous>)
	at a (/Users/joel/.vscode/extensions/natizyskunk.sftp-1.15.0/dist/extension.js:61:583)
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:93:5) when remote ➞ local /Users/joel/formtest.log
```

**Comment by Natizyskunk** [comment]
please use older 1.14.0 version since i'm working on a fix.

@kripper, can you please take a look at it ? It seems to come from the implementation you added in the `scr/core/transfertTask.ts` file.

**Comment by kripper** [comment]
Yes, I'll fix it ASAP.

**Comment by mrjcgoodwin** [comment]
It may be coincidence as only tried a few projects but I think the issue may only be occurring on projects using a password in the settings file. Project I'm working on at the moment uses an SSH key and seems ok.

**Comment by Mr-KayJayDee** [comment]
> It may be coincidence as only tried a few projects but I think the issue may only be occurring on projects using a password in the settings file. Project I'm working on at the moment uses an SSH key and seems ok.

That is not true, because i have two project on the same server, both using the same config, one works, the other don't.

**Comment by mrjcgoodwin** [comment]
> > It may be coincidence as only tried a few projects but I think the issue may only be occurring on projects using a password in the settings file. Project I'm working on at the moment uses an SSH key and seems ok.
> 
> That is not true, because i have two project on the same server, both using the same config, one works, the other don't.

Ha - ok then please disregard!

**Comment by jjnxpct** [comment]
Is als get this error when I use 'reidt in local' of when downloading a folder:

[error] TypeError [ERR_INVALID_ARG_TYPE]: The "fd" argument must be of type number. Received undefined

I have installed the latest version 1.15.1 (Windows 10)

The files / folder does get downloaded, so everything seems to work OK, but I do get this warning.

**Comment by Natizyskunk** [comment]
Should be fixed with next update and new 1.15.2 version.
